backend/apps/mentorship/management/commands/link_issues_by_label.py

```diff
@@ -0,0 +1,63 @@
+"""A command to link issues to mentorship modules based on matching labels."""
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from apps.github.models.issue import Issue
+from apps.mentorship.models.module import Module
+
+
+class Command(BaseCommand):
+    """Efficiently link issues to mentorship modules based on matching labels."""
+
+    help = "Efficiently link issues to mentorship modules based on matching labels."
+
+    def handle(self, *_args, **options) -> None:
+        """Handle the command execution.
+
+        Args:
+            *_args: Variable length argument list.
+            **options: Arbitrary keyword arguments containing command options.
+
+        """
+        self.stdout.write("Fetching all issues and labels...")
+        # load id
+        issues = Issue.objects.prefetch_related("labels").only("id")
+
+        label_to_issue_ids: dict[str, set[int]] = {}
+        for issue in issues:
+            for label in issue.labels.all():
+                label_to_issue_ids.setdefault(label.name, set()).add(issue.id)
```

Comment:
Label name is not unique. The same name label can exist in multiple repositories.